plugins/modules/files/sapcar.py

```diff
@@ -0,0 +1,233 @@
+      path: "~/source/hana.sar"
+      manifest: "MyNewSignature.SMF"
+      signature: true
+    register: sapcar
+
+"""
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+
+module = None
+
+
+def getListOfFiles(dirName):
+    # create a list of file and directories
+    # names in the given directory
+    listOfFile = os.listdir(dirName)
+    allFiles = list()
+    # Iterate over all the entries
+    for entry in listOfFile:
+        # Create full path
+        fullPath = os.path.join(dirName, entry)
+        # If entry is a directory then get the list of files in this directory
+        if os.path.isdir(fullPath):
+            allFiles = allFiles + [fullPath]
+            allFiles = allFiles + getListOfFiles(fullPath)
+        else:
+            allFiles.append(fullPath)
+    return allFiles
+
+
+def downloadSAPCAR(binary_path):
+    bin_path = None
+    # download sapcar binary if url is provided other wise path is returned
+    try:
+        if not os.path.isfile(binary_path):
+            with open_url(binary_path) as response, open("/tmp/sapcar", 'wb') as out_file:
+                data = response.read()
+                out_file.write(data)
+            os.chmod("/tmp/sapcar", 755)
+            bin_path = "/tmp/sapcar"
+
+    finally:
+        if bin_path is None:
+            os.chmod(binary_path, 755)
+            bin_path = binary_path
+
+    return bin_path
+
+
+def checkifPresent(command, path, dest, signature, manifest):
+    # manipuliating output from SAR file for compare with already extracted files
+    iter_command = [command, '-tvf', path]
+    sar_out = module.run_command(iter_command)[1]
+    sar_raw = sar_out.split("\n")[1:]
+    if dest[-1] != "/":
+        dest = dest + "/"
+    sar_files = [dest + x.split(" ")[-1] for x in sar_raw if x]
+    # remove any SIGNATURE.SMF from list because it will not unpacked if signature is false
+    if not signature:
+        sar_files = [item for item in sar_files if '.SMF' not in item]
+    # if signature is renamed manipulate files in list of sar file for compare.
+    if manifest != "SIGNATURE.SMF":
+        sar_files = [item for item in sar_files if '.SMF' not in item]
+        sar_files = sar_files + [manifest]
+    # get extracted files if present
+    files_extracted = getListOfFiles(dest)
+    # compare extracted files with files in sar file
+    present = all(elem in files_extracted for elem in sar_files)
+    return present
+
+
+def main():
+    global module
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(type='path', required=True),
+            dest=dict(type='path'),
+            binary_path=dict(type='path'),
+            signature=dict(type='bool', default=False),
+            manifest=dict(type='str', default="SIGNATURE.SMF"),
+            remove=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+    )
+    rc, out, err = ["", "", ""]
```

Comment:
I would not use `rc` for a string, but only for an integer (return code of the command).